src/main/java/org/terasology/launcher/packages/PackageDatabase.java

```diff
@@ -159,4 +170,40 @@ String getUrl() {
             return trackedPackages;
         }
     }
+
+    private static class RepositoryDeserializer implements JsonDeserializer<Repository> {
+        @Override
+        public Repository deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
```

Comment:
can this be done based on or supported by the schema.json somehow?

Reply:
I'm afraid there's no direct way to do that. Actually, I couldn't find a way to integrate json-schema with Gson. So the file needs to be parsed twice to get the things done. :sweat_smile: